tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java

```diff
@@ -1810,8 +1869,24 @@ private String handleCurrencyFormat(XPathParts parts, String value) {
         }
 
         double sampleAmount = 1295.00;
-        String example = formatNumber(df, sampleAmount);
-        example = addExampleResult(formatNumber(df, -sampleAmount), example);
+        example = addExampleResult(formatNumber(df, sampleAmount), example, showContexts);
+        example = addExampleResult(formatNumber(df, -sampleAmount), example, showContexts);
+        
+        if (showContexts && !altAlpha && countValue == null) {
```

Comment:
suggestion for future improvement: a block this size warrants adding a subroutine